refactor(osemgrep): Move `cli_output` conversion out of Output.ml

nmote opened the pull request:
Now the conversion logic is in a separate module, called from `Scan_subcommand.ml`. This is in preparation for porting over autofix, which will need the fix that is computed during the `cli_match` conversion.

Test plan: make

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

aryx commented on semgrep-core/src/osemgrep/cli_scan/Output.ml:
```diff
@@ -13,256 +13,11 @@ module Out = Semgrep_output_v0_j
 
-let output_result (conf : Scan_CLI.conf) (res : Core_runner.result) : unit =
-  let (cli_output : Out.cli_output) = cli_output_of_core_results conf res in
+let output_result (conf : Scan_CLI.conf) (cli_output : Out.cli_output) : unit =
```
I'm not sure we want to change the type of output_result. It was fine getting Core_runner.result because there might
be more information in it at some point than in Out.cli_output.

aryx commented on semgrep-core/src/osemgrep/cli_scan/Scan_subcommand.ml:
```diff
@@ -68,7 +68,11 @@ let run (conf : Scan_CLI.conf) : Exit_code.t =
   in
 
-  Output.output_result conf res;
+  let cli_output : Out.cli_output =
```
I would have left that in Output.ml.

aryx reviewed (approved):
Missing Cli_output.mli
(we need to make a semgrep rule for it ... it's not expressive enough for it though).

nmote committed:
refactor(osemgrep): Move `cli_output` conversion out of Output.ml

Now the conversion logic is in a separate module, called from
`Scan_subcommand.ml`. This is in preparation for porting over autofix,
which will need the fix that is computed during the `cli_match`
conversion.

Test plan: make

aryx commented on semgrep-core/src/osemgrep/cli_scan/Cli_json_output.mli:
```diff
@@ -0,0 +1,7 @@
+val cli_output_of_core_results :
+  Scan_CLI.conf -> Core_runner.result -> Semgrep_output_v0_j.cli_output
+
+val lines_of_file :
```
maybe some comments about lines_of_file, or use Common.filename instead of string.

nmote replied:
https://github.com/returntocorp/semgrep/pull/6399
